pkg/remote/trans/netpoll/http_client_handler_test.go

```diff
@@ -17,48 +17,179 @@
+	httpCilTransHdlr, _ = NewHTTPCliTransHandlerFactory().NewTransHandler(cilopt)
+}
+
+// TestHTTPWrite test http_client_handler Write return err
+func TestHTTPWrite(t *testing.T) {
+	// 1. prepare mock data
+	conn := &MockNetpollConn{}
+	rwTimeout := time.Second
+	cfg := rpcinfo.NewRPCConfig()
+	rpcinfo.AsMutableRPCConfig(cfg).SetReadWriteTimeout(rwTimeout)
+	to := rpcinfo.NewEndpointInfo("", "", nil, map[string]string{
+		"http_url": "https://example.com",
+	})
+	ri := rpcinfo.NewRPCInfo(nil, to, nil, cfg, rpcinfo.NewRPCStats())
+	ctx := context.Background()
+	msg := remote.NewMessage(nil, mocks.ServiceInfo(), ri, remote.Reply, remote.Client)
+
+	// 2. test
+	err := httpCilTransHdlr.Write(ctx, conn, msg)
+	// check err not nil
+	test.Assert(t, err != nil)
+}
+
+// TestHTTPRead test http_client_handler Read return err
+func TestHTTPRead(t *testing.T) {
+	// 1. prepare mock data
+	rwTimeout := time.Second
+
+	var readTimeout time.Duration
+	var isReaderBufReleased bool
+	conn := &MockNetpollConn{
+		SetReadTimeoutFunc: func(timeout time.Duration) (e error) {
+			readTimeout = timeout
+			return nil
+		},
+		ReaderFunc: func() (r netpoll.Reader) {
+			reader := &MockNetpollReader{
+				ReleaseFunc: func() (err error) {
+					isReaderBufReleased = true
+					return nil
+				},
+			}
+			return reader
+		},
+	}
+
+	cfg := rpcinfo.NewRPCConfig()
+	rpcinfo.AsMutableRPCConfig(cfg).SetReadWriteTimeout(rwTimeout)
+	ri := rpcinfo.NewRPCInfo(nil, nil, nil, cfg, rpcinfo.NewRPCStats())
+	ctx := context.Background()
+	msg := remote.NewMessage(nil, mocks.ServiceInfo(), ri, remote.Reply, remote.Client)
+
+	// 2. test
+	err := httpCilTransHdlr.Read(ctx, conn, msg)
+	// check err not nil
+	test.Assert(t, err != nil)
+
+	httpCilTransHdlr.OnError(ctx, err, conn)
+	test.Assert(t, readTimeout == trans.GetReadTimeout(ri.Config()))
+	test.Assert(t, isReaderBufReleased)
+}
+
+// TestHTTPPanicAfterRead test http_client_handler OnMessage success
+func TestHTTPOnMessage(t *testing.T) {
+	// 1. prepare mock data
+	svcInfo := mocks.ServiceInfo()
+
+	ri := rpcinfo.NewRPCInfo(nil, nil, rpcinfo.NewInvocation(svcInfo.ServiceName, method), nil, rpcinfo.NewRPCStats())
+	ctx := rpcinfo.NewCtxWithRPCInfo(context.Background(), ri)
+	recvMsg := remote.NewMessageWithNewer(svcInfo, ri, remote.Call, remote.Server)
+	sendMsg := remote.NewMessage(svcInfo.MethodInfo(method).NewResult(), svcInfo, ri, remote.Reply, remote.Server)
+
+	// 2. test
+	_, err := httpCilTransHdlr.OnMessage(ctx, recvMsg, sendMsg)
+	test.Assert(t, err == nil, err)
+}
+
+// TestAddMetaInfo test http_client_handler addMetaInfo success
+func TestAddMetaInfo(t *testing.T) {
+	// 1. prepare mock data
+	cfg := rpcinfo.NewRPCConfig()
+	ri := rpcinfo.NewRPCInfo(nil, nil, nil, cfg, rpcinfo.NewRPCStats())
+	var req interface{}
+	msg := remote.NewMessage(req, mocks.ServiceInfo(), ri, remote.Reply, remote.Client)
+	h := http.Header{}
+
+	// 2. test
+	err := addMetaInfo(msg, h)
```

Comment:
there should be a check for the http header to ensure that `addMetaInfo` works correctly